### PR TITLE
Quaser IFE challenger

### PR DESCRIPTION
**Pull request by kevsul**
Closes https://github.com/omgnetwork/plasma-contracts/issues/733

### CODE REVIEW

File: ife-challenger/src/contracts/SpendingConditionRegistry.json [hunk 1]
```diff
@@ -0,0 +1,3080 @@
+{
```
Comment (souradeep-das) [inline ife-challenger/src/contracts/SpendingConditionRegistry.json:1]: do we need this abi?

**Review by souradeep-das — APPROVED** [review]
Works great! 👍  added some minor comments.

File: ife-challenger/src/challenger.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      const ifes = result.filter((e) => e.event === 'non_canonical_ife');
+
+      try {
+        /* eslint-disable no-await-in-loop */
+        for (let i = 0; i < ifes.length; ++i) {
+          const ife = ifes[i];
+
+          // Check if any of the ifes are quasar ifes
+          const { inputs } = transaction.decodeTxBytes(ife.details.txbytes);
+          for (let j = 0; j < ifes.length; ++j) {
```
[the remark this answers is not in the document]
Comment (kevsul) [reply ife-challenger/src/challenger.js:27]: Good catch! 😅 
Yeah, checking every input may not be strictly necessary, but I wanted to be sure that if the ife claim input is spent in _any_ tx at all then it should invalidate the ife claim.

File: ife-challenger/src/contracts/SpendingConditionRegistry.json (hunk 1 above)
Comment (kevsul) [reply ife-challenger/src/contracts/SpendingConditionRegistry.json:1]: Nope. Left over from testing

File: ife-challenger/src/challenger.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (souradeep-das) [reply ife-challenger/src/challenger.js]: sounds good!

**Review by souradeep-das — APPROVED** [review]

**Pull request merged**